File: datasets/rb/HumanEval_29_filter_by_prefix.rb
# Filter an input list of strings only for ones that start with a given prefix.
# >>> filter_by_prefix([], 'a')
# []
# >>> filter_by_prefix(['abc', 'bcd', 'cde', 'array'], 'a')
# ['abc', 'array']
def filter_by_prefix(strings, prefix)
  strings.select { |string| string.start_with?(prefix) }
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_filter_by_prefix
    candidate = method(:filter_by_prefix)
    assert_equal([], candidate.call([], 'john'))
    assert_equal(['xxx', 'xxxAAA', 'xxx'], candidate.call(['xxx', 'asd', 'xxy', 'john doe', 'xxxAAA', 'xxx'], 'xxx'))
  end
end

File: datasets/rs/HumanEval_29_filter_by_prefix.rs
/// Filter an input list of strings only for ones that start with a given prefix.
/// >>> filter_by_prefix([], 'a')
/// []
/// >>> filter_by_prefix(['abc', 'bcd', 'cde', 'array'], 'a')
/// ['abc', 'array']
fn filter_by_prefix(strings: Vec<String>, prefix: String) -> Vec<String> {
    strings
        .into_iter()
        .filter(|s| s.starts_with(&prefix))
        .collect()

}

fn main() {
    let candidate = filter_by_prefix;
    assert_eq!(candidate(Vec::<String>::new(), String::from("john")), Vec::<String>::new());
    assert_eq!(candidate(vec![String::from("xxx"), String::from("asd"), String::from("xxy"), String::from("john doe"), String::from("xxxAAA"), String::from("xxx")], String::from("xxx")), vec![String::from("xxx"), String::from("xxxAAA"), String::from("xxx")]);
}
